File: 4295-count-indices-with-opposite-parity/count-indices-with-opposite-parity.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var countOppositeParity = function (nums) {
    const n = nums.length;
    const ans = new Array(n).fill(0);

    let even = 0;

    for (let x of nums) {
        if (x % 2 === 0) {
            even++;
        }
    }

    let odd = n - even;

    for (let i = 0; i < n; i++) {
        if (nums[i] % 2 === 0) {
            even--;
            ans[i] = odd;
        } else {
            odd--;
            ans[i] = even;
        }
    }

    return ans;
};

File: 4295-count-indices-with-opposite-parity/count-indices-with-opposite-parity.py
class Solution:
    def countOppositeParity(self, nums: list[int]) -> list[int]:
        n = len(nums)
        ans = [0] * n

        even = sum(1 for x in nums if x % 2 == 0)
        odd = n - even

        for i in range(n):
            if nums[i] % 2 == 0:
                even -= 1
                ans[i] = odd
            else:
                odd -= 1
                ans[i] = even

        return ans
